### ISSUE TITLE
[a11y]: Violations found on StructuredList component in v10 and v11

### ISSUE THREAD

**Issue by LauKwaiFanHK** [carbon-design-system/carbon#11986]
### Package

@carbon/react

### Browser

Chrome, Firefox

### Operating System

MacOS

### Package version

v10, v11

### Automated testing tool and ruleset

IBM Accessibility Checker - Latest Deployment

### Description

I used IBM Accessibility Checker to scan a newly developed page based on React which uses the StructuredList component. When I scan this component independently (the page on storybook), I found that the StructuredList components violates 4.1.2 Name, Role, Value (v10) and 4.1.2 Name, Role, Value (v11). 

### WCAG 2.1 Violation

Success Criterion 4.1.2 (Name, Role, Value) for v10; 2.4.1 Bypass Blocks for v11

### Reproduction/example

Violations are found on Carbon-React storybook for v10 and v11

### Steps to reproduce

To reproduce violations on v10:
1. Go to https://v10-react.carbondesignsystem.com/iframe.html?id=components-structuredlist--selection&viewMode=story&args= 
2. Scan this page using IBM Accessibility Checkers
3. 20 violations of 4.1.2 Name, Role, Value will be found:
![Screenshot 2022-08-22 at 14 41 26](https://user-images.githubusercontent.com/81176980/185923534-54a6a154-9d4b-4f28-8baf-7fa6f137e455.png)

To reproduce violations on v11:
1. Go to https://react.carbondesignsystem.com/iframe.html?viewMode=story&id=components-structuredlist--selection&args=  
2. Scan this page using IBM Accessibility Checkers
3. 5 violations of 2.4.1 Bypass Blocks will be found
![Screenshot 2022-08-22 at 14 42 01](https://user-images.githubusercontent.com/81176980/185923715-0f7fad1a-0e8e-4f44-a5a8-dd7e277f4bb5.png)
:

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by abbeyhrt** [comment]
Hey @LauKwaiFanHK! The violations you found in v10 are something that I will address but just FYI, those violations in the v11 storybook are okay and don't need to be addressed :)  Since it is just a storybook example page, and not a real application use of our components, we do not provide a landmark element since cannot to specify for users as our components will be used within many different landmarks. 

We also can't provide a "skip to main content" on the pages the individual component pages storybook creates since,  whatever is there is the only content on the page and therefore a "Skip to content" would be superfluous.

**Comment by abbeyhrt** [comment]
Looking into the violations found in v10, they unfortunately cannot be addressed meaningfully without a breaking change. The violations didn't exist at the time but the reason we refactored StructuredList in v11 is due to the accessibility concerns that the violations are now pointing out. If anyone is running into these violations in their product, we would recommend using the v11 version of StructuredList. It is possible to use just one component from v11 if a complete migration is too difficult at this time.

**Comment by abbeyhrt** [comment]
Closing this due to above explanations, if anyone has anymore questions about this PR, feel free to ping the Carbon team!
